src/assembly/datasource.ts

```diff
@@ -0,0 +1,45 @@
+import * as host from "./hypermode";
+import { JSON } from "json-as";
+
+
+@json
+class pgInput {
+  query!: string;
+  params!: string[];
+}
+
+
+@json
+class pgResponse {
+  result!: string;
+  error!: string;
+}
+
+export abstract class postgresql {
+  static query(
+    datasourceName: string,
+    query: string,
+    params: string[],
+  ): string {
```

Comment:
```suggestion
  static query<TData>(
    datasourceName: string,
    query: string,
    params: QueryParameters | null = null,
  ): TData[] {
```

see feedback on the examples.